src/python/test/test_dxabs.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2013-2015 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import os, unittest, tempfile, filecmp, time, json, sys, shutil
+import string
+import subprocess
+
+import requests
+from requests.packages.urllib3.exceptions import SSLError
+
+import dxpy
+import dxpy_testutil as testutil
+from dxpy.exceptions import (DXAPIError, DXFileError, DXError, DXJobFailureError, ResourceNotFound)
+from dxpy.utils import pretty_print, warn
+from dxpy.utils.resolver import resolve_path, resolve_existing_path, ResolutionError, is_project_explicit
+
+def remove_all(proj_id, folder="/"):
+    dxproject = dxpy.DXProject(proj_id)
+    dxproject.remove_folder(folder, recurse=True)
+
+def setUpTempProjects(thing):
+    thing.old_workspace_id = dxpy.WORKSPACE_ID
+    thing.proj_id = dxpy.api.project_new({'name': 'azure-test-project', 'region': 'azure:westus'})['id']
+    dxpy.set_workspace_id(thing.proj_id)
+
+def tearDownTempProjects(thing):
+    dxpy.api.project_destroy(thing.proj_id, {'terminateJobs': True})
+    dxpy.set_workspace_id(thing.old_workspace_id)
+
+class TestDXFile(unittest.TestCase):
```

Comment:
I'd like these tests to only run when the user specifically calls for them to be run, since not all users will have Azure regions enabled. I added some infrastructure to support this on master. Please merge it into your branch and then:

* Make this test class conditional on testutil.TEST_AZURE (using unittest.skipUnless)
* The region should be set based on testutil.TEST_AZURE instead of hard-coded here (in case the user has access to some azure region other than westus.